// profanity-scanner/profanity_scanner.js
var fs = require('fs')

function Profanity_Scanner() {

    this.badwords = fs.readFileSync('/Users/mi0152/IdeaProjects/NodeApi/node_modules/profanity-scanner/badwords.csv').toString().split(",")

    this.scan = function(text, done) {
        // remove meaningless chars and pad with spaces to make detection better
        text.replace(/[-_=\+\/\\\.<>\?\*;\(\)\{\}:]/g, ' ')
        text = ' ' + text + ' '

        var profane = false
        this.badwords.forEach(function(word) {
            if (~text.indexOf(' ' + word + ' ')) {
                profane = true
            }
        })

        return profane
    }

    this.add_word = function(word) {
        this.badwords.push(word)
        return save_words(this.badwords)
    }

    this.remove_word = function(word) {
        var index = this.badwords.indexOf(word)
        if (index != -1) {
            this.badwords.splice(index, 1)
            return save_words(this.badwords)
        } else {
            return false
        }
    }

    function save_words(badwords) {
        try {
            fs.writeFileSync('badwords.csv', badwords.join())
            return true
        } catch (e) {
            return false
        }
    }

}

module.exports = Profanity_Scanner

if (require.main == module) {

    var pf = new Profanity_Scanner()
    var assert = require('assert')
    assert(pf.scan('hello there!') == false, 'detects simple good language')
    assert(pf.scan('fuck off') == true, 'detects simple bad language')

}
